django/thunderstore/social/providers.py

```diff
@@ -0,0 +1,112 @@
+from typing import Dict
+
+import requests
+from django.conf import settings
+
+
+class BaseOauthHelper:
+    token = None
+
+    def __init__(self, code: str, redirect_url: str) -> None:
+        self.code = code
+        self.redirect_url = redirect_url
+
+    def complete_login(self) -> None:
+        """
+        Fetch access token with the temporary code and configured secret.
+        """
+        data = {
+            "client_id": self.CLIENT_ID,
+            "client_secret": self.CLIENT_SECRET,
+            "code": self.code,
+            "grant_type": "authorization_code",  # Required by Discord
+            "redirect_uri": self.redirect_url,  # Required by Discord
```

Comment:
given that the OAuth spec calls for `redirect_uri` being the naming, should we change the var name here to match? (this isn't a Discord specific naming scheme)

Reply:
Renamed.